sycl/include/sycl/usm.hpp

```diff
@@ -19,280 +20,248 @@ namespace sycl {
 ///
 // Explicit USM
 ///
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const device &dev, const context &ctxt,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const device &dev, const context &ctxt,
-    const property_list &propList,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const queue &q,
-    const detail::code_location CodeLoc = detail::code_location::current());
-__SYCL_EXPORT void *malloc_device(
-    size_t size, const queue &q, const property_list &propList,
-    const detail::code_location CodeLoc = detail::code_location::current());
+__SYCL_EXPORT void *malloc_device(size_t size, const device &dev,
+                                  const context &ctxt _CODELOCPARAM(&CL));
```

Comment:
A bit of a nitpick but arguably `CodeLoc` was a better name as it better conveys the intention and hopefully has less chance to conflict with macros.
I realize we had a bunch of `CL` parameters before so it is not make-or-break, but we might as well improve it while we are in here. 😄

Reply:
Thanks for feedback but I think `CL` is sufficiently wrapped with more descriptive macros like `_CODELOCPARAM` to be a NIT. Do you agree?

Reply:
Good point. I suppose it is, except for the cases where they are passed on. That only leaves the macro collision problem, but in the end that is already a topic being discussed, so as long as `CL` isn't a commonly-used macro I am okay with it for now.